docs/IE_DG/network_state_intro.md

```diff
@@ -0,0 +1,193 @@
+					<dim>1</dim>
+				</port>
+			</input>
+			<output>
+				<port id="2" precision="FP32">
+					<dim>1</dim>
+					<dim>1</dim>
+				</port>
+			</output>
+		</layer>
+		<layer id="5" name="output/sink_port_0" type="Result" version="opset5">
+			<input>
+				<port id="0">
+					<dim>1</dim>
+					<dim>1</dim>
+				</port>
+			</input>
+		</layer>
+	</layers>
+	<edges>
+		<edge from-layer="0" from-port="1" to-layer="1" to-port="0"/>
+                <edge from-layer="2" from-port="0" to-layer="3" to-port="1"/>
+                <edge from-layer="1" from-port="1" to-layer="3" to-port="0"/>
+                <edge from-layer="3" from-port="2" to-layer="4" to-port="0"/>
+                <edge from-layer="3" from-port="2" to-layer="10" to-port="0"/> 
+                <edge from-layer="1" from-port="1" to-layer="10" to-port="1"/>
+                <edge from-layer="10" from-port="2" to-layer="5" to-port="0"/>
+	</edges>
+	<meta_data>
+		<MO_version value="unknown version"/>
+		<cli_parameters>
+		</cli_parameters>
+	</meta_data>
+</net>
+```
+
+### Example of creating model Ngraph API
+
+```cpp
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{1, 1});
+    auto init_const = op::Constant::create(element::f32, Shape{1, 1}, {0});
+    auto read = make_shared<op::ReadValue>(init_const, "v0");
+    std::vector<shared_ptr<Node>> args = {arg, read};
+    auto add = make_shared<op::Add>(arg, read);
+    auto assign = make_shared<op::Assign>(add, "v0");
+    auto add2 = make_shared<op::Add>(add, read);
+    auto res = make_shared<op::Result>(add2);
+
+    auto f = make_shared<Function>(ResultVector({res}), ParameterVector({arg}), SinkVector({assign}));
+```
+
+In this example `SinkVector` was used to create Ngraph::Function. For network with states except inputs and outputs also Assign nodes should be pointed to Function 
+to avoid deleting it during graph transformations. It can be done with constructor as shown in example or with special method `Fuction::add_sink`. Also you can delete 
+sink from `Function` after deleting node from graph with `Function::delete_sink` method.
+
+## OpenVINO state API
+
+ InferenceEngine have method `InferRequest::QueryState` to get list of states from network and IVariableState interface to operate with state. Brief description of methods 
+ is below and next section contains small workable example how this interface can be used.
+ 
+ * `std::string GetName() const`
+   returns name(variable_id) of according Variable
+ * `void Reset()`
+   reset state to default value
+ * `void SetState(Blob::Ptr newState)`
+   set new value for state
+ * `Blob::CPtr GetState() const`
+   returns current value of state
+
+## Example of inference network with state
+
+Let's take IR from example in prevous sections. In this example inference of 2 independent sequences of data will be demonstrated. Between these sequences state 
+should be reset.
+
+@snippet openvino/docs/snippets/InferenceEngine_network_with_state_infer.cpp part1
```

Comment:
`openvino/docs/` should be removed. 

You can compile the documentation using `make openvino_docs` and make sure that your document looks well.